DataFormats/GeometryCommonDetAlgo/interface/GlobalErrorBase.h

```diff
@@ -85,15 +119,56 @@ class GlobalErrorBase
   T czz() const {
     return theCartesianError(2,2);
   }
+
+  T ctx() const {
+    return theCartesianError(3,0);
+  }
   
- /**
+  T cty() const {
+    return theCartesianError(3,1);
+  }
+  
+  T ctz() const {
+    return theCartesianError(3,2);
+  }
+
+  T ctt() const {
+    return theCartesianError(3,3);
+  }
+  
+  /**
    * Access method to the matrix,
    * /return The SymMatrix
    */
-  const AlgebraicSymMatrix33 & matrix() const {
+  const AlgebraicSymMatrix33 matrix() const {
+    AlgebraicSymMatrix33 out;
+    out(0,0) = theCartesianError(0,0);
+    out(1,0) = theCartesianError(1,0);
+    out(1,1) = theCartesianError(1,1);
+    out(2,0) = theCartesianError(2,0);
+    out(2,1) = theCartesianError(2,1);
+    out(2,2) = theCartesianError(2,2);
+    return out;
+  }
+  const AlgebraicSymMatrix33 matrix_new() const {
+    AlgebraicSymMatrix33 out;
+    out(0,0) = theCartesianError(0,0);
+    out(1,0) = theCartesianError(1,0);
+    out(1,1) = theCartesianError(1,1);
+    out(2,0) = theCartesianError(2,0);
+    out(2,1) = theCartesianError(2,1);
+    out(2,2) = theCartesianError(2,2);
+    return out;
+  }
+
+ /**
+   * Access method to the matrix,
+   * /return The SymMatrix 4x4
+   */
+  const AlgebraicSymMatrix44& matrix4D() const {
     return theCartesianError;
   }
-  const AlgebraicSymMatrix33 & matrix_new() const {
+  const AlgebraicSymMatrix44& matrix_new4D() const {
```

Comment:
If there is no purpose to the `matrix4D/matrix_new4D` duplication, why do it? We only need `matrix4D`.